aioauth/response_type.py

```diff
@@ -151,12 +160,12 @@ async def create_authorization_response(
             request,
             client.client_id,
             request.query.scope,
-            request.query.response_type,  # type: ignore
+            request.query.response_type,
             request.query.redirect_uri,
-            request.query.nonce,  # type: ignore
+            request.query.nonce,
         )
 
-        return IdTokenResponse(id_token=id_token)
+        return IdTokenResponse(id_token=id_token or "")
```

Comment:
Any thoughts on what else might be done here? Most other things want `get_id_token` to be optional, but that's obviously not the case here.

Reply:
I would keep this line as before:

```suggestion
        return IdTokenResponse(id_token=id_token)
```

Reply:
I guess it's also fine to make `TokenResponse.id_token` optional even if `Storage.get_id_token` isn't optional, but seems kind of deceiving. Anyway, I leave the choice to you.

Reply:
In regards to the IdTokenResponse, it is essential that it remains mandatory. However, we can add an additional condition that if the id_token is None, then an internal server error exception should be raised.

When it comes to the grant_types, the id_token is optional for TokenResponse, at least in Auth0. For instance, Auth0 does not include the id_token for the client_credentials grant type. You can refer to this link for more information: https://auth0.com/docs/get-started/authentication-and-authorization-flow/call-your-api-using-the-client-credentials-flow#response.

Although I am uncertain which solution would be the most appropriate and compliant with standards, according to this document, https://openid.net/specs/openid-connect-core-1_0.html#TokenResponse, the id_token is mandatory.

Reply:
I responded to the feature flag notion indirectly over here: https://github.com/aliev/aioauth/pull/75#issuecomment-1483900865. I think what I said there applies here too. So, I'll try something like that for some of this also.